vignettes/migration.Rmd

```diff
@@ -0,0 +1,111 @@
+vignette: >
+  %\VignetteIndexEntry{Migration from didehpc}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+The `hipercow` package shares many of its ideas from [`didehpc`](https://mrc-ide.github.io/didehpc/) but is a new implementation, sharing little code and offering no direct migration path.  If you have an existing `didehpc`-based project, you should keep using that and use `hipercow` for any new work.
+
+If you have never used `didehpc` before, there is no need to read this vignette.
+
+# Differences
+
+Some differences of note between `hipercow` and `didehpc`; if you are confused by some other difference not mentioned here please let us know.
+
+**The user interface is quite different, with free functions replacing the big `queue_didehpc` object**.  Previously you created an object (often `obj`) with [`queue_didehpc`](https://mrc-ide.github.io/didehpc/reference/queue_didehpc.html) and then interact with this by running a method (say `obj$task_result()`); instead in `hipercow` you use a free function (in this example `task_result`). See below for a translation of methods to new function(s).
+
+**No object corresponding to a task**. Previously there were object handles that had methods. So you had a task `t` and fetched its result as `t$result()`. As with the queue, this has moved to free functions (in this example `task_result()`).
+
+**We no longer check that your packages have been installed**. We might modify this behaviour, but the previous behaviour of not allowing the queue object to be created until packages have been installed has been stopped.  Instead, we provide some tools (notably, `hipercow_provision_check()`) for you to check that the state of the packages is what you expect them to be. There's nothing to stop you submitting jobs into a cluster environment that cannot run them due to missing packages, but we hope that will be more flexible.
+
+**We never load packages into your current environment, you don't even need them installed**. In didehpc when you loaded the queue, it loaded all packages in your context in your current session. This no longer happens, and the packages do not need to be installed at all.
+Smaller differences include:
+
+* `task_wait` does not return the result of the task, but a boolean indicating if the task was successful
+
+## Mapping of `didehpc` methods to `hipercow` functions
+
+### `queue_didehpc`
+
+* `cluster_load`: TODO (`mrc-4892`)
+* `config`: `hipercow_configuration()`
+* `context`: no analogue
+* `dide_id`: no analogue (we might add this to `task_info` though)
+* `dide_log`: `task_log_show(<id>, outer = TRUE)`
+* `enqueue`: `task_create_expr()`
+* `enqueue_`: `task_create_explicit`
+* `enqueue_bulk`: `task_create_bulk_expr`
+* `initialize_context`: no analogue
+* `install_packages`: `hipercow_provision(method = "pkgdepends", refs = ...)`
+* `lapply`: TODO (`mrc-4895`)
+* `login`: `windows_check` (we don't actually log in this way though)
```

Comment:
I think this was `web_login` ?

Reply:
Not within the class, but there was a `web_login` free function that I think you're thinking of here